workers/unity/Packages/com.improbable.gdk.querybasedinterest/InterestQuery.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    public class InterestQuery
+    {
+        private static readonly List<uint> EmptyList = new List<uint>();
+
+        private ComponentInterest.Query query;
+
+        public static InterestQuery Query(ComponentInterest.QueryConstraint constraint)
+        {
+            var interest = new InterestQuery
+            {
+                query =
+                {
+                    Constraint = constraint,
+                    FullSnapshotResult = true,
+                    ResultComponentId = EmptyList
+                }
+            };
+            return interest;
+        }
+
+        public InterestQuery MaxFrequencyHz(float frequency)
+        {
+            query.Frequency = frequency;
+            return this;
+        }
+
+        public ComponentInterest.Query Filter(params uint[] resultComponentIds)
```

Comment:
`FilterResults` ?

Reply:
Something to suggest this selects the components that you want to receive as a result of the query